azure-quantum/azure/quantum/job/job.py

```diff
@@ -142,21 +142,212 @@ def get_results(self, timeout_secs: float = DEFAULT_TIMEOUT):
                 if "Histogram" not in results:
-                    raise f"\"Histogram\" array was expected to be in the Job results for \"{self.details.output_data_format}\" output format."
+                    raise ValueError(f"\"Histogram\" array was expected to be in the Job results for \"{self.details.output_data_format}\" output format.")
                 
                 histogram_values = results["Histogram"]
 
                 if len(histogram_values) % 2 == 0:
                     # Re-mapping {'Histogram': ['[0]', 0.50, '[1]', 0.50] } to {'[0]': 0.50, '[1]': 0.50}
                     return {histogram_values[i]: histogram_values[i + 1] for i in range(0, len(histogram_values), 2)}
                 else: 
-                    raise f"\"Histogram\" array has invalid format. Even number of items is expected."
+                    raise ValueError(f"\"Histogram\" array has invalid format. Even number of items is expected.")
+            elif self.details.output_data_format == "microsoft.quantum-results.v2":
+                if "DataFormat" not in results or results["DataFormat"] != "microsoft.quantum-results.v2":
+                    raise ValueError(f"\"DataFormat\" was expected to be \"microsoft.quantum-results.v2\" in the Job results for \"{self.details.output_data_format}\" output format.")
+
+                if "Results" not in results:
+                    raise ValueError(f"\"Results\" field was expected to be in the Job results for \"{self.details.output_data_format}\" output format.")
+                
+                if len(results["Results"]) < 1:
+                    raise ValueError("\"Results\" array was expected to contain at least one item")
+                
+                results = results["Results"][0]
+
+                if "Histogram" not in results:
+                    raise ValueError(f"\"Histogram\" array was expected to be in the Job results for \"{self.details.output_data_format}\" output format.")
+
+                if "Shots" not in results:
+                    raise ValueError(f"\"Shots\" array was expected to be in the Job results for \"{self.details.output_data_format}\" output format.")
+
+                histogram_values = results["Histogram"]
+
+                total_count = len(results["Shots"])
+
+                # Re-mapping object {'Histogram': [{"Outcome": [0], "Display": '[0]', "Count": 500}, {"Outcome": [1], "Display": '[1]', "Count": 500}]} to {'[0]': 0.50, '[1]': 0.50}
+                return {outcome["Display"]: outcome["Count"] / total_count for outcome in histogram_values}
             
             return results
         except:
             # If errors decoding the data, return the raw payload:
             return payload
 
+    def get_results_histogram(self, timeout_secs: float = DEFAULT_TIMEOUT):
+        """Get job results histogram (in V2 format) by downloading the results blob from the
+        storage container linked via the workspace.
```

Comment:
```suggestion
        """Get job results histogram by downloading the results blob from the
        storage container linked via the workspace.
```

Reply:
The v2 format is an implementation detail so it's better to avoid it in doc comments.